aea/cli/generate_key.py

```diff
@@ -64,7 +70,7 @@ def _generate_private_key(type_: str, file: Optional[str] = None) -> None:
             PRIVATE_KEY_PATH_SCHEMA.format(type__) if file is None else file
         )
         if _can_write(private_key_file):
-            create_private_key(type__, private_key_file)
+            create_private_key(type__, private_key_file, password)
```

Comment:
We might want to add a password confirmation step here!

Reply:
We might also want to disable autogeneration of keys outside the MAM context. As otherwise the password confirmation cannot be done. In the MAM context, we assume the password is correctly set.